tests/Microsoft.Identity.Test.Unit/ExceptionTests/MsalExceptionTests.cs

```diff
@@ -509,6 +509,56 @@ await app.AcquireTokenForClient(TestConstants.s_scope)
             }
         }
 
+        [TestMethod]
+        public async Task AuthorityInNotFoundExceptions()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                harness.HttpManager.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.AuthorityCommonTenant + TestConstants.DiscoveryEndPoint));
+                harness.HttpManager.AddMockHandlerContentNotFound(HttpMethod.Post);
+
+                ConfidentialClientApplication app = null;
+                var certificate = new X509Certificate2(
+                    ResourceHelper.GetTestResourceRelativePath("RSATestCertDotNet.pfx"));
+                
+                    app = ConfidentialClientApplicationBuilder
+                        .Create(TestConstants.ClientId)
+                        .WithAuthority(new System.Uri(ClientApplicationBase.DefaultAuthority), true)
+                        .WithRedirectUri(TestConstants.RedirectUri)
+                        .WithHttpManager(harness.HttpManager)
+                        .WithCertificate(certificate)
+                        .BuildConcrete();
+                
+
+                var ex = await Assert.ThrowsExceptionAsync<MsalServiceException>(async () =>
+                {
+                    await app.AcquireTokenForClient(TestConstants.s_scope)
+                             .WithExtraQueryParameters("extra=qp")
+                             .ExecuteAsync(CancellationToken.None)
+                             .ConfigureAwait(false);
+                }).ConfigureAwait(false);
+
+                Assert.IsFalse(string.IsNullOrEmpty(ex.CorrelationId));
+                Assert.IsTrue(ex.Message.Contains("Authority used: https://login.microsoftonline.com/common/oauth2/v2.0/token"));

```

Comment:
That's not the authority, that's the token endpoint.

Reply:
Yeah, I need to revise the code.